docs/source/interface.rst

```diff
@@ -14,34 +14,35 @@ Calling Interface
 ###################
 
 A transition defined in a smart contract can be called either by the issuance
-of a transaction or by message calls from another smart contract. As such, the
+of a transaction or by message calls from another smart contract. The same
 calling interface will be used to call the contract via external transactions
 and inter-contract message calls.
 
-The inputs to the interpreter (``scilla-runner``) consists of four JSON files
-as described below. Every execution of the interpreter can be provided with
-these four json inputs: ::
+The inputs to the interpreter (``scilla-runner``) consists of four input JSON
+files as described below. Every invocation of the interpreter to execute a 
+transition must be provided with these four json inputs: ::
 
     ./scilla-runner -init init.json -istate input_state.json -iblockchain input_blockchain.json -imessage input_message.json -o output.json -i input.scilla
 
 The interpreter executable can be run either to create a contract (denoted
-``CreateContract``) or to invoke a function in a contract (``InvokeContract``).
-Depending on which of these two cases, some of the arguments will be absent.
-The table below presents the arguments that should be present in either of the
-two cases.  A ``CreateContract`` can be distinguished from an
-``InvokeContract``, based on the presence of ``input_message.json``. If the
-argument is absent, then the interpreter will evaluate it as a
-``CreateContract``. Else, it will treat it as an ``InvokeContract``. Note that
-for ``CreateContract``, the interpreter only performs basic type checking to
-match with contract’s immutable parameters.
+``CreateContract``) or to invoke a transition (function) in a contract (``InvokeContract``).
+Depending on which of these two, some of the arguments will be absent.
+The table below outlays the arguments that should be present in eac of
+these two cases.  A ``CreateContract`` is distinguished from an
+``InvokeContract``, based on the presence of ``input_message.json`` and
+``input_state.json``. If these argument is absent, then the interpreter will 
```

Comment:
these argument is -> these arguments are

Reply:
fixed